src/components/DonationWidget/machine/index.ts

```diff
@@ -0,0 +1,180 @@
+        city: '',
+        zipCode: '',
+        country: ''
+      },
+      cardInformation: {
+        firstName: '',
+        lastName: '',
+        email: '',
+        cardNumber: ''
+      }
+    },
+    initial: 'amountForm',
+    states: {
+      amountForm: {
+        initial: 'donateOnce',
+        states: {
+          donateOnce: {
+            entry: ['setOnceDonation'],
+            on: {
+              NEXT: '#donation.paymentForm',
+              'START.MONTHLY': 'donateMonthly',
+              'UPDATE.AMOUNT': {
+                target: 'donateOnce',
+                actions: ['updateDonationOnceAmount']
+              }
+            }
+          },
+          donateMonthly: {
+            entry: ['setMonthlyDonation'],
+            on: {
+              NEXT: '#donation.paymentForm',
+              'START.ONCE': 'donateOnce',
+              'UPDATE.AMOUNT': {
+                target: 'donateMonthly',
+                actions: ['updateDonationMonthlyAmount']
+              }
+            }
+          },
+          hist: {
+            type: 'history',
+            history: 'shallow'
+          }
+        }
+      },
+      paymentForm: {
+        initial: 'cardForm',
+        states: {
+          cardForm: {
+            on: {
+              NEXT: [
+                {
+                  target: 'addressForm',
+                  cond: 'paymentFormCompleted',
+                  actions: ['updateCardInformation']
+                }
+              ],
+              PREV: '#donation.amountForm.hist',
+              'START.ONCE': '#donation.amountForm.donateOnce',
+              'START.MONTHLY': '#donation.amountForm.donateMonthly'
+            }
+          },
+          addressForm: {
+            on: {
+              NEXT: [
+                {
+                  target: '#donation.processDonation',
+                  cond: 'addressFormCompleted',
+                  actions: ['updateBillingInformation']
+                }
+              ],
+              PREV: 'cardForm',
+              'START.ONCE': '#donation.amountForm.donateOnce',
+              'START.MONTHLY': '#donation.amountForm.donateMonthly'
+            }
+          }
+        }
+      },
+      processDonation: {
+        invoke: {
+          id: 'submitDonation',
+          src: 'donationService',
+          onDone: {
+            target: 'success',
+            actions: assign({ donation: (context, event) => event.data })
+          },
+          onError: {
+            target: 'failure',
+            actions: assign({ error: (context, event) => event.data })
+          }
+        }
+      },
+      success: {
+        type: 'final'
+      },
+      failure: {
+        on: {
+          RETRY: { target: '#donation.paymentForm.cardForm' }
+        }
+      }
+    }
+  },
+  {
+    guards: {
+      paymentFormCompleted: (context, event) => {
+        const { firstName, lastName, email, cardNumber } = event;
+        return firstName && lastName && email && cardNumber;
```

Comment:
Since these are strings, what do you think about the following?

```suggestion
paymentFormCompleted: (context, event) => {
  const { firstName, lastName, email, cardNumber } = event;
  return [firstName, lastName, email, cardNumber].every(Boolean);
}
```

Reply:
I think it is sexier hahahaha. Will do, but won't commit here to do the same in all.